AIObject.py

```diff
@@ -0,0 +1,41 @@
+from abc import ABC, abstractmethod
+
+from openai import OpenAI
+
+
+class AbstractAIO(ABC):
+    ai_client = None
+    ai_model = None
+    ai_prompt = None
+
+    def __init__(self, ai_client,  ai_model=None, ai_prompt=None):
+        self.ai_client = ai_client
+        self.ai_model = ai_model
+        self.ai_prompt = ai_prompt
+        super().__init__()
+
+    @abstractmethod
+    def get_ai_response(self, text):
+        pass
+
+
+
+class OpenAIClient(AbstractAIO):
```

Comment:
Could this not just be a function that takes the `ai_service` from the yaml as a dictionary input? 

(Not a problem if the Enterprise OO constructs like abstract base classes and factories are earning their keep, or you think will be soon, but ideally all the tooling should be as simple to understand and modify as possible)

Reply:
Mmm. That's a good point. 🤔

I could just pass in the whole `ai_service`. 

I'll fix it.

Reply:
I'm going to keep the structure as is for the time being at least, until I've had a chance to play with other AI setups. This does keep a lot of the complexity in the factory. (That's the plan anyway)